### DISCUSSION TITLE
q Toggle button's thumb -slider part (toggle order) is in between of toggle button  in ui.table

### DISCUSSION THREAD

**Discussion by Shivar90** [zauberzeug/nicegui#2437] [Q&A]
### Question

I am using Nicegui version 1.4.11 when I get data from excel In Table and table is created on UI there are toggle buttons in each row but toggle when display in each row toggle button thumb (sliding part) appears in between of toggle button .Refer snap. But when I write code to create table using columns and rows I am getting toggle button thumb correctly. Please explain Nicegui community  what is the issue. Refer q button toggle code.

```Python
df = pd.read_excel("data_output.xlsx")
    print("df_data", df)
    table = ui.table(columns=columns, rows=df.to_dict('records'), row_key='name', pagination=5).classes('w-full')
    Columns_Name = table.columns
    print("Columns Name:", Columns_Name)
    if any(column['label'] == 'Action' for column in table.columns):
        print(f"The table contains the 'Action' column.")
        table.add_slot('body-cell-action', '''
            <q-td key="action" :props="props">
                <q-toggle 
                    :color="props.value ? 'red' : 'green'" 
                    keep-color
                    :label="props.value ? 'Acquired' : 'Available'" 
                    checked-icon="clear" 
                    v-model="props.row.action" 
                    v-model="check_value" 
                    @update:model-value="() => $parent.$emit(
                        <!-- callback function -->
                        'action_event', <!-- should match string passed to `table.on` below -->
                        
                        <!-- arguments, can be whatever you want really -->
                        props.row,   <!-- this would pass the whole row of data -->
                        props.row.Place,   <!-- this would pass data of the Place column for this row -->
                        props.value ? 'Acquire' : 'Release'  <!-- this will pass a string indicating the action based on the toggle switch state -->
                    )"
                />           
            </q-td>
        ''')
        table.on('action_event', handle_action_event)

def handle_action_event(e: events.GenericEventArguments) -> None:
    print(f"{e.sender}")  # this will be the table object
    print(f"{e.args}")  # this will be the args defined in the emit call
    ui.notify(f'{e.args[1]} wants action {e.args[2]}')
```
![image](https://github.com/zauberzeug/nicegui/assets/36196330/d56aee95-6e30-4783-8676-fa941f7e5fbb)

**Comment by falkoschindler** [comment]
Hi @Shivar90,

The toggle value is initialized with `check_value`, which doesn't appear anywhere in the code. If it is undefined, it explains the toggle button in the middle. What is `check_value` supposed to be?

**Reply by Shivar90** [reply]
@falkoschindler Thanks for instant reply. check_value is not used anywhere in code (previously different logic kept for that). Even I removed " check_value" in code still working same even used :toggle order ='f' as mentioned in documentation "https://quasar.dev/vue-components/toggle#qtoggle-api" still get same observation of toggle button .please tell how to keep toggle button in default state(false) . Also I am unable to set toggle button when refresh screen (functionality based on condition if Acquired :toggle button should set "true" and if Available: toggle should set false state of toggle button) .please give idea how to set toggle button state based on REST API value

**Reply by falkoschindler** [reply]
Here is a rough example for toggle elements that synchronize with the server via emitted events:

```py
columns = [
    {'name': 'name', 'label': 'Name', 'field': 'name'},
    {'name': 'vote', 'label': 'Vote', 'field': 'vote'},
]
rows = [
    {'name': 'Alice', 'vote': True},
    {'name': 'Bob', 'vote': False},
    {'name': 'Charlie', 'vote': True},
]

def handle_vote(e: events.GenericEventArguments) -> None:
    name, vote = e.args
    for row in rows:
        if row['name'] == name:
            row['vote'] = vote
            table.update()

table = ui.table(columns=columns, rows=rows, row_key='name')
table.add_slot('body-cell-vote', r'''
    <q-td :props="props">
        <q-toggle v-model="props.value" @update:model-value="(value) => $parent.$emit('vote', props.row.name, value)" />
    </q-td>
''')
table.on('vote', handle_vote)
```

**Reply by Shivar90** [reply]
@falkoschindler Thanks for this code able to save the toggle button status even when refresh webpage
